test/integration/role_test.lua

```diff
@@ -0,0 +1,66 @@
+local fio = require('fio')
+local t = require('luatest')
+local helpers = require('test.helper')
+local g = t.group('expirationd_intergration_role')
+local is_cartridge_helpers, cartridge_helpers = pcall(require, 'cartridge.test-helpers')
+
+g.before_all(function(cg)
+    if is_cartridge_helpers then
+        local entrypoint_path = fio.pathjoin(helpers.project_root,
+                                             'test',
+                                             'entrypoint',
+                                             'srv_role.lua')
+        cg.cluster = cartridge_helpers.Cluster:new({
+            datadir = fio.tempdir(),
+            server_command = entrypoint_path,
+            use_vshard = true,
+            replicasets = {
+                {
+                    uuid = helpers.uuid('a'),
+                    alias = 'router',
+                    roles = { 'vshard-router' },
+                    servers = {
+                        { instance_uuid = helpers.uuid('a', 1), alias = 'router' },
+                    },
+                },
+                {
+                    uuid = helpers.uuid('b'),
+                    alias = 's-1',
+                    roles = { 'vshard-storage', 'customers-storage', 'expirationd' },
+                    servers = {
+                        { instance_uuid = helpers.uuid('b', 1), alias = 's1-master' },
+                    }
+                }
+            },
+        })
+        cg.cluster:start()
+    end
+end)
+
+g.after_all(function(cg)
+    if is_cartridge_helpers then
+        cg.cluster:stop()
+        fio.rmtree(cg.cluster.datadir)
+    end
+end)
+
+g.before_each(function()
+    t.skip_if(not is_cartridge_helpers, "cartridge is not installed")
+end)
+
+-- init/stop/validate_config/apply_config tested in test/unit/role_test.lua
```

Comment:
To be honest, I would've preferred to have a simple "expire some tuples with task started purely with cartridge configuration" integration test.

Reply:
I've added an integration test with starting tasks from a config.